src/vmm/tests/integration_tests.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+#![cfg(target_arch = "x86_64")]
+
+extern crate libc;
+extern crate vmm;
+
+use std::convert::TryFrom;
+use std::io;
+use std::path::PathBuf;
+use std::thread::sleep;
+use std::time::Duration;
+
+use libc::{_exit, fork, waitpid, WEXITSTATUS, WIFEXITED};
+
+use vmm::{KernelConfig, MemoryConfig, VMMConfig, VcpuConfig, VMM};
+use vmm_sys_util::terminal::Terminal;
+
+fn default_memory_config() -> MemoryConfig {
+    MemoryConfig { mem_size_mib: 1024 }
+}
+
+fn default_kernel_config() -> KernelConfig {
+    let kernel_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+        .join("../../resources/kernel/vmlinux-hello-busybox-halt");
+    // Sanity check. Because the VMM runs in a separate process, if the file doesn't exist,
+    // all we see is a different exit code than 0.
+    assert!(kernel_path.as_path().exists());
```

Comment:
While reviewing https://github.com/rust-vmm/rust-vmm-ci/pull/37, I noticed that even if this path doesn't exist, so the assert fails/should fail, the test is still passing. I assume this is not the desired behavior. Can you expand a bit on how is this exit code propagated further?

Reply:
Great catch! The default panic handler exited with 0, so the parent process thought the child was on the happy path. I moved the `assert` before the `fork` to avoid the unnecessary new process on the error path, and installed a panic hook that exits a panicky process with a nonzero exit code.